examples/plot_wlasso.py

```diff
@@ -5,95 +5,118 @@
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
-    X_train, X_val, y_train, y_val = train_test_split(
-        X_train, y_train, test_size=0.5)
+##############################################################################
+# Dataset creation
+##############################################################################
+# X, y = make_regression(n_samples=600, n_features=600, noise=60, random_state=2)
+n_samples = 600
+n_features = 600
+rng = check_random_state(0)
+X = rng.multivariate_normal(
+    size=n_samples, mean=np.zeros(n_features),
+    cov=toeplitz(0.5 ** np.arange(n_features)))
+
+
+# Create true regression coefficients of 5 non-zero values
+w_true = np.zeros(n_features)
+size_supp = 5
+idx = rng.choice(
+    X.shape[0], size_supp, replace=False)
+w_true[idx] = (-1) ** np.arange(size_supp)
+noise = rng.randn(n_samples)
+y = X @ w_true
+y += noise / norm(noise) * 0.5 * norm(y)
+
+# Here we split the dataset (X, y) in 3:
+# the regression coefficients will be determined using X_train, y_train
+# the regularization parameter will be calibrated using X_val, y_val
+# the model is then tested on unseen data X_test, y_test
+X_train_val, X_test, y_train_val, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=2)
+X_train, X_val, y_train, y_val = train_test_split(
+    X_train_val, y_train_val, test_size=0.5, random_state=2)
+##############################################################################
 
-n_samples, n_features = X_train.shape
 
-print("Starting path computation...")
-n_samples = len(y_train)
+# max penalty value
 alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
-log_alpha0 = np.log(alpha_max / 10)
+n_alphas = 30  # number of iter in the line search, ie 30 evals of the gradient
+alphas = alpha_max * np.geomspace(1, 0.001, n_alphas)
 
-n_alphas = 10
-p_alphas = np.geomspace(1, 0.0001, n_alphas)
-alphas = alpha_max * p_alphas
-log_alphas = np.log(alphas)
-
-tol = 1e-7
-max_iter = 1e5
 
 ##############################################################################
-# Grid-search
-# -----------
+# vanilla LassoCV
+print("========== Celer's LassoCV started ===============")
+model_cv = LassoCV(
+    verbose=False, fit_intercept=False, alphas=alphas, tol=1e-7, max_iter=100,
+    cv=2).fit(X_train_val, y_train_val)
+
+# measure mse on test
+mse_cv = mean_squared_error(y_test, model_cv.predict(X_test))
+print("Vanilla LassoCV: Mean-squared error on test data %f" % mse_cv)
+##############################################################################
 
-# The sklearn solver is indeed very long on the considered problems!
-# estimator = linear_model.Lasso(
-#     fit_intercept=False, max_iter=1000, warm_start=True)
 
-# celer is way faster !
-# https://github.com/mathurinm/celer
+##############################################################################
+# Weighted Lasso with sparse-ho
+# We use the vanilla lassoCV coefficients as a starting point
```

Comment:
this does not render fine see https://245-252772479-gh.circle-artifacts.com/0/dev/auto_examples/plot_wlasso.html#sphx-glr-auto-examples-plot-wlasso-py